docs/workflows/best-practices-osh.md

```diff
@@ -0,0 +1,398 @@
+2. _Do not add project or organization specific headers._ 
+    > [LB: to what??? to the source code of libraries or frameworks?]​
+3. _Use only a single version of each library or framework​._: Also, do not have copies of the same library installed. 
+   Note that in some cases, you can have a library _L1_ that requires _M_, and a library _L2_ that requires another version of _M_; in such cases you may not be able to influence this (depending on what your package manager allows), but at least ensure your application code does not directly rely on multiple versions of the same library.
+[Note Asma] Point 3 sounds like transitive dependency management, what are the best practices there to handle those in a package manager? Also at which lvl of transitivity do we stop caring? --> indeed in some package managers you actually do have influence over the versions of indirect dependencies, discuss this. exercising influence over this makes sense for security purposes, but otherwise its impact *should* be encapsulated.
+
+#### Regression tests and maintainability of the application code are key to updating frameworks/libraries​
+If it is hard to update a library, chances are the problem lies in your codebase.​
+1. _Keep module coupling and component independence low_, to make it easier to change code implementation (such as dealing with new versions of a library) while keeping the same behavior/requirements.
+1. _Develop, maintain and run regression tests._ These help to identify breaking changes in updates.​​
+[Note Asma] aren't breaking changes mentioned in the update notes? Analysing those might be a good first step to identify those before upgrading.
+
+
+### 2. Define OSH related policies for development
+There are a number of policies on how to address open source libraries during development. For most of these policies, minimal requirements should be set for all teams; individual teams may agree on more stringent rules. 
+
+1. _Define the usage of a package manager_: choose the package manager(s) to be used, at least per system, preferably shared across the organization. Depending on the technologies that are used, you may need multiple package managers.
+   - The package managers need to be integrated in your CI/CD pipeline.
+  <!-- more useful info here: https://gurukuldevops.com/package-management-in-devops-tools-and-best-practices/ -->   
+
+2. _Set the thresholds for library risks_ that are (not) acceptable: this is applicable to all types of risks. Set these goals in the [Sigrid objectives](#TBD). Define goals for:
+   - The amount of and allowed types of vulnerabilities.
+   - The allowed licenses.
+   - Library freshness; the time since a new version of a library has been released.
+
+[Note Asma] Amount of vulnerabilites is not always a fitting metric. Having a goal to reach 0 critical and or high risk findings makes sense. Low and mediums are often present in larger numbers 
```

Comment:
For security yes, but in my experience this is not the case for OSH. Most of the vulnerable libs we see are either 'high' or 'medium', with some 'low's sprinkled in. Reporting OSH vulnerabilities is relatively expensive with CVEs being issued and everything, so I guess that's not worth it for 'low' vulnerabilities.

Reply:
As a result, I rarely find myself actually using the vulnerability rating when making assessments. It's basically, is there a fix -> apply it. If not, understand why and take it from there. Most maintainers try to address open CVEs because people start asking questions about it.